src/Tasks/Microsoft.NET.Build.Tasks/ResourceUpdater.cs

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Provides methods for modifying the embedded native resources
+    /// in a PE image. It currently only works on Windows, because it
+    /// requires various kernel32 APIs.
+    /// </summary>
+    public class ResourceUpdater
+    {
+        //
+        // Native methods for updating resources
```

Comment:
Standard practice has been to put all P/Invokes and associated code in a private sealed class. It then allows use of the `nameof` operator and scopes functions properly. Unsure how important that is in this code base.

```
private sealed class Kernel32
{
        [DllImport(nameof(Kernel32), SetLastError=true)]
        private static extern IntPtr BeginUpdateResource(string pFileName,
                                                         [MarshalAs(UnmanagedType.Bool)]bool bDeleteExistingResources);
} 
```

Reply:
There have been a lot of different "standard" practices over the years. I'm happy to follow whatever is current in coreclr/corefx. We don't really have any interop in this repo yet apart from tests. I am deferring to you guys on how we should be doing it.

---
In reply to: [219338255](https://github.com/dotnet/sdk/pull/2545#discussion_r219338255) [](ancestors = 219338255)

Reply:
The class named after module + nameof is nice, but I don't feel strongly about it.

---
In reply to: [219643850](https://github.com/dotnet/sdk/pull/2545#discussion_r219643850) [](ancestors = 219643850,219338255)

Reply:
I used your suggested convention.